topdown/http_test.go

```diff
@@ -2233,6 +2233,148 @@ func TestHTTPSNoClientCerts(t *testing.T) {
 	})
 }
 
+func TestCertSelectionLogic(t *testing.T) {
+	const (
+		localCaFile = "testdata/ca.pem"
+	)
+
+	// Set up Environment
+	caCertPEM, err := ioutil.ReadFile(localCaFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	caPool := x509.NewCertPool()
+	if ok := caPool.AppendCertsFromPEM(caCertPEM); !ok {
+		t.Fatal("failed to parse CA cert")
+	}
+
+	ca, err := ioutil.ReadFile(localCaFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = os.Setenv("CLIENT_CA_ENV", string(caCertPEM))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	getClientTlsConfig := func(obj ast.Object) *tls.Config {
+		_, client, err := createHTTPRequest(BuiltinContext{Context: context.Background()}, obj)
+		if err != nil {
+			t.Fatalf("Unexpected error creating HTTP request %v", err)
+		}
+		if client.Transport == nil {
+			return nil
+		}
+		return client.Transport.(*http.Transport).TLSClientConfig
+	}
+
+	systemCertsPool, err := x509.SystemCertPool()
+	if err != nil {
+		t.Fatalf("Unexpected error reading system certs %v", err)
+	}
+
+	tempSystemCertsPool, err := x509.SystemCertPool()
+	if err != nil {
+		t.Fatalf("Unexpected error reading system certs %v", err)
+	}
+	systemCertsAndCaPool, err := addCACertsFromBytes(tempSystemCertsPool, ca)
+	if err != nil {
+		t.Fatalf("Unexpected error merging system certs and ca certs %v", err)
+	}
+
+	tests := []struct {
+		note     string
+		input    map[*ast.Term]*ast.Term
+		expected [][]byte
+		err      string
```

Comment:
I'd rename this to `msg` or something as `err` is an error type by convention.

Reply:
Good idea, I renamed it to `msg` as you suggested